pkg/apis/kubeone/v1beta1/types.go

```diff
@@ -0,0 +1,324 @@
+/*
+Copyright 2020 The KubeOne Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"encoding/json"
+
+	"github.com/kubermatic/kubeone/pkg/apis/semver"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// KubeOneCluster is KubeOne Cluster API Schema
+type KubeOneCluster struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Name is the name of the cluster
+	Name string `json:"name"`
+	// Hosts describes the control plane nodes and how to access them
+	Hosts []HostConfig `json:"hosts,omitempty"`
+	// StaticWorkers allows the user to define a list of nodes as workers that are not managed by MachineController
+	StaticWorkers []HostConfig `json:"staticWorkers,omitempty"`
+	// APIEndpoint are pairs of address and port used to communicate with the Kubernetes API
+	APIEndpoint APIEndpoint `json:"apiEndpoint,omitempty"`
+	// CloudProvider configures the cloud provider specific features
+	CloudProvider CloudProviderSpec `json:"cloudProvider,omitempty"`
+	// Versions defines which Kubernetes version will be installed
+	Versions VersionConfig `json:"versions,omitempty"`
+	// ClusterNetwork configures the in-cluster networking
+	ClusterNetwork ClusterNetworkConfig `json:"clusterNetwork,omitempty"`
+	// Proxy configures proxy used while installing Kubernetes and by the Docker daemon
+	Proxy ProxyConfig `json:"proxy,omitempty"`
+	// Workers is used to create worker nodes using the Kubermatic machine-controller
+	Workers []WorkerConfig `json:"workers,omitempty"`
```

Comment:
```suggestion
	DynamicWorkers []WorkerConfig `json:"dynamicWorkers,omitempty"`
```

Albeit, I'm not 100% sure about this name.

Reply:
Sounds good enough to me.